packages/valory/skills/check_stop_trading_abci/behaviours.py

```diff
@@ -0,0 +1,162 @@
+
+    @mech_request_count.setter
+    def mech_request_count(self, mech_request_count: int) -> None:
+        """Set the liveness period."""
+        self._mech_request_count = mech_request_count
+
+    def _get_mech_request_count(self) -> WaitableConditionType:
+        """Get the mech request count."""
+        status = yield from self.contract_interact(
+            contract_address=self.params.mech_contract_address,
+            contract_public_id=MechContract.contract_id,
+            contract_callable="get_requests_count",
+            data_key="requests_count",
+            placeholder=get_name(CheckStopTradingBehaviour.mech_request_count),
+            address=self.synchronized_data.safe_contract_address,
+        )
+        return status
+
+    @property
+    def is_first_period(self) -> bool:
+        """Return whether it is the first period of the service."""
+        return self.synchronized_data.period_count == 0
+
+    @property
+    def params(self) -> CheckStopTradingParams:
+        """Return the params."""
+        return cast(CheckStopTradingParams, self.context.params)
+
+    def is_staking_kpi_met(self) -> WaitableConditionType:
+        """Return whether the staking KPI has been met (only for staked services)."""
+        yield from self.wait_for_condition_with_sleep(self._check_service_staked)
+        self.context.logger.info(f"{self.service_staking_state=}")
+        if self.service_staking_state != StakingState.STAKED:
+            return False
+
+        yield from self.wait_for_condition_with_sleep(self._get_mech_request_count)
+        mech_request_count = self.mech_request_count
+        self.context.logger.info(f"{self.mech_request_count=}")
+
+        yield from self.wait_for_condition_with_sleep(self._get_service_info)
+        mech_request_count_on_last_checkpoint = self.service_info[2][1]
+        self.context.logger.info(f"{mech_request_count_on_last_checkpoint=}")
+
+        yield from self.wait_for_condition_with_sleep(self._get_ts_checkpoint)
+        last_ts_checkpoint = self.ts_checkpoint
+        self.context.logger.info(f"{last_ts_checkpoint=}")
+
+        yield from self.wait_for_condition_with_sleep(self._get_liveness_period)
+        liveness_period = self.liveness_period
+        self.context.logger.info(f"{liveness_period=}")
+
+        yield from self.wait_for_condition_with_sleep(self._get_liveness_ratio)
+        liveness_ratio = self.liveness_ratio
+        self.context.logger.info(f"{liveness_ratio=}")
+
+        mech_requests_since_last_cp = mech_request_count - mech_request_count_on_last_checkpoint
+        self.context.logger.info(f"{mech_requests_since_last_cp=}")
+
+        current_timestamp = self.synced_timestamp
+        self.context.logger.info(f"{current_timestamp=}")
+
+        required_mech_requests = math.ceil(max(
+            (current_timestamp - last_ts_checkpoint) * liveness_ratio / 10**18,
+            (liveness_period) * liveness_ratio / 10**18
+        )) + 1
```

Comment:
Please specify constants for `10**18` and `1`.

Reply:
Fixed.